src/routers/task.js

```diff
@@ -65,30 +67,71 @@ router.patch('/tasks/:id', auth, async(req, res) => {
     const allowedUpdates = ['title', 'completed']
     const isValidOperation = updates.every((update) => allowedUpdates.includes(update))
     if (!isValidOperation) {
-        return res.status(400).send({ error: 'Property doesn\'t exist.' })
+        return res.status(400).json({ error: 'Property doesn\'t exist.' })
     }
     try {
         const task = await Task.findOne({ _id: req.params.id, createdBy: req.user._id })
         if (!task) {
-            return res.status(404).send()
+            return res.status(404).json()
         }
         updates.forEach((update) => task[update] = req.body[update])
         await task.save()
-        res.send(task)
+        res.json(task)
     } catch (error) {
-        res.status(400).send(error)
+        res.status(400).json(error)
     }
 })
 
 router.delete('/tasks/:id', auth, async(req, res) => {
     try {
         const task = await Task.findOneAndDelete({ _id: req.params.id, createdBy: req.user._id })
         if (!task) {
-            return res.status(404).send()
+            return res.status(404).json()
         }
-        res.status(200).send(task)
+        res.status(200).json(task)
     } catch (error) {
-        res.status(400).send(error)
+        res.status(400).json(error)
+    }
+})
+
+const upload = multer({
+    limits: {
+        fileSize: 1000000
+    },
+    fileFilter(req, file, callback) {
+        if (!file.originalname.match(/\.(jpg|jpeg|png)$/)) {
+            return callback(new Error('Please upload an image document.'))
+        }
+        callback(undefined, true)
+    }
+})
+
+router.post('/tasks/:id/logo', auth, upload.single('logo'), async(req, res) => {
+    try {
+        const buffer = await sharp(req.file.buffer).resize({ width: 250, height: 250 }).png().toBuffer()
+        const update = { logo: buffer }
+        const task = await Task.findOneAndUpdate({ _id: req.params.id, createdBy: req.user._id }, update, { new: true })
+        if (!task) {
+            return res.status(404).json()
+        }
+        res.status(200).json(task)
+    } catch (error) {
+        res.status(500).json(error)
+    }
+}, (error, req, res, next) => {
```

Comment:
I'd rather apply this line to all endpoints in the project 
see **The default error handler**
https://expressjs.com/en/guide/error-handling.html

Reply:
Get rid this block

Reply:
Fixed.